staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go

```diff
@@ -445,7 +445,8 @@ const (
 	// deletes all the dependents whose ownerReference.blockOwnerDeletion=true
 	// from the key-value store.  API sever will put the "foregroundDeletion"
 	// finalizer on the object, and sets its deletionTimestamp.  This policy is
-	// cascading, i.e., the dependents will be deleted with Foreground.
+	// cascading, i.e., the dependents will be deleted with Foreground.  This
+	// policy is non-blocking.
```

Comment:
What do you mean by "non-blocking" here?

Reply:
I mean that the method call returns before the deletion is necessarily finished. One might expect a "foreground" deletion request to not come back (send a response) until the "foreground" deletion completes. However, that is not the case here.

Reply:
No delete calls are "blocking" in that sense in Kubernetes.

What Foreground deletion "blocks" is the removal of the object in question. That object will be held with a finalizer until all referencing objects are deleted.

That is, if you DELETE with Foreground an object, then immediately GET it, the object will still be there. The object will not finish the deletion process for some time. And until that happens, you will not be able to create another object of the same name.

Reply:
> That is, if you DELETE with Foreground an object, then immediately GET it, the object will still be there. The object will not finish the deletion process for some time. And until that happens, you will not be able to create another object of the same name.

This is not the behavior I expected when I read the Python API documentation as currently written, which is why I'm wanting it to be documented here.

Reply:
I'd rather explain with something like "After beginning the deletion, you can wait for this to complete by WATCHing the object you requested deletion of."